file/spring-cloud-starter-stream-source-file/src/main/java/org/springframework/cloud/stream/app/file/source/FileSourceConfiguration.java

```diff
@@ -66,7 +66,7 @@
 	Source source;
 
 	@Bean
-	public TriggerProperties triggerProperties() {
+	public static TriggerProperties triggerProperties() {
```

Comment:
This generally feels like a bit of a smell for me because too much is going on inside the `@Configuration` class.  Decoupling this by putting the creation inside another class seems like a cleaner solution than introducing a static.